File: sip-task-core/src/main/kotlin/com/sip/task/core/annotation/ScheduleTask.java
package com.sip.task.core.annotation;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author tianhu
 * @date 2019/7/24
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD})
@Documented
@Inherited
public @interface ScheduleTask {
    String code() default "";
}

File: sip-task-core/src/main/kotlin/com/sip/task/core/annotation/ScheduleTask.kt
package com.sip.task.core.annotation

import java.lang.annotation.Documented
import java.lang.annotation.Inherited
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * @author tianhu
 * @date 2019/7/24
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Documented
@Inherited
annotation class ScheduleTask(val code: String = "")
